src/pages/WalletPage.vue

```diff
@@ -352,10 +378,84 @@ export default {
       "checkPendingTokens",
       "decodeRequest",
       "generateNewMnemonic",
+      "redeem",
     ]),
     ...mapActions(useCameraStore, ["closeCamera", "showCamera"]),
     ...mapActions(useNWCStore, ["listenToNWCCommands"]),
     ...mapActions(useNPCStore, ["generateNPCConnection", "claimAllTokens"]),
+    knowThisMintOfTokenJson: function (tokenJson) {
+      const mintStore = useMintsStore();
+      // check if we have all mints
+      for (var i = 0; i < tokenJson.token.length; i++) {
+        if (
+          !mintStore.mints.map((m) => m.url).includes(token.getMint(tokenJson))
+        ) {
+          return false;
+        }
+      }
+      return true;
+    },
```

Comment:
this function is a verbatim copy from `ReceiveTokenDialog.vue` - isn't there a nicer way how to reuse the code rather than copy-pasting it?

Reply:
Yea like I said in the description of this PR I could put that in a "store". Probably in the `receiveTokensStore` and then map the action to ReceiveTokenDialog, SendTokenDialog, WalletPage

Reply:
Better now?

Reply:
Yes!